src/megatron/energon/epathlib/epath.py

```diff
@@ -74,6 +75,24 @@ def __init__(
             else:
                 raise ValueError(f"Unknown protocol: {protocol}")
 
+    @staticmethod
+    def _resolve(path: Union[str, PurePosixPath]) -> PurePosixPath:
+        if isinstance(path, str):
+            path = PurePosixPath(path)
+        parts = path.parts
+        if parts[0] != "/":
+            raise ValueError("Only absolute paths are supported")
+        if ".." in parts or '.' in parts:
+            new_parts = []
+            for part in parts:
+                if part == '..':
+                    new_parts.pop()
+                else:
```

Comment:
Since we are expecting `.` parts in the path (see `if` above), suggest to add an `elif` branch to handle `.` parts and remove them.

Reply:
Right, added it